### ISSUE TITLE
useContext return undefine when ContextProvider lazy rendering

### ISSUE THREAD

**Issue by mduclehcm** [ryansolid/solid#243]
I have created codesandbox to reproduce
```js
const HistoryContext = createContext();

function App() {
  return (
    <ContextProvider value={{ msg: "hello" }}>
      {() => <Consume />} {/* OK */}
      <Consume /> {/* context is undefined */}
    </ContextProvider>
  );
}

function ContextProvider(props) {
  return (
    <HistoryContext.Provider value={props.value}>
      {props.children}
    </HistoryContext.Provider>
  );
}

function useHistory() {
  return useContext(HistoryContext);
}

function Consume() {
  const history = useHistory();
  return <h1>typeof history:{typeof history}</h1>;
}
```

https://codesandbox.io/s/eager-kilby-155hz?file=/src/ssr.js

**Comment by ryansolid** [comment]
Thanks for opening. The new server runtime is much less tested it's likely the simplified model isn't handling the case properly (most methods are shallow approximations of their reactive counterparts). I will investigate.

**Comment by ryansolid** [comment]
I was wrong it is the compiler. I need to keep property lazy evaluation in SSR. For some reason I was thinking otherwise. But it's always possible to mess up context.

**Comment by ryansolid** [comment]
Fixed in v0.21.0
